### ISSUE TITLE
Are RMSE and MSE losses the same during training? 

### ISSUE THREAD

**Issue by nd7141** [microsoft/LightGBM#5260]
In [documentation of LightGBM](https://lightgbm.readthedocs.io/en/latest/Parameters.html#objective) it says that MSE and RMSE loss functions are the same aliases for 'regression' task:
![image](https://user-images.githubusercontent.com/4607709/171600620-66d7e0b8-72e2-47b6-9f1f-e47203bb7d07.png)

However, to me these are different functions that lead to different optimization. 

Here is MSE: 
![image](https://user-images.githubusercontent.com/4607709/171600857-7f9b3743-b734-4875-854b-a1a6844ebb86.png)

And here is RMSE: 
![image](https://user-images.githubusercontent.com/4607709/171601042-10df84c5-f6b1-41e3-a74b-ff18f22da8ac.png)

The thing is that for this formulation of RMSE, the second-order gradient, i.e. hessian would be a matrix because the gradient would depend on all predictions from all observations. While for MSE second-order gradient would be a vector. So it should be much easier to compute MSE than RMSE. And, of course, this should lead to completely different predictions. 

So I just try to understand is it that for RMSE you just compute MSE loss or you actually do compute RMSE loss as indicated above? 

I can see that [in the code](https://github.com/microsoft/LightGBM/blob/1c558a54e99c0081df1402b5384e1921ff74ebdf/src/objective/regression_objective.hpp#L113) you have `sqrt` variable, but I'm not sure what it does.

**Comment by jmoralez** [comment]
Hi @nd7141. Given that the square root is a monotonic function, minimizing the MSE yields the same solution as minimizing the RMSE (though the objective function values differ). And `root_mean_squared_error` is indeed just an alias for `mean_squared_error`: https://github.com/microsoft/LightGBM/blob/f1328d5c5fee1da5907371e292dda3e8344b4b63/include/LightGBM/config.h#L1176-L1179
so that objective minimizes the MSE.

The `sqrt` variable you reference is actually [`reg_sqrt`](https://lightgbm.readthedocs.io/en/latest/Parameters.html#reg_sqrt) which trains on the square root of the label and squares the predictions.

Please let us know if this helps.
